File: e-elearning-main/service/internal/entity/chapter.go
package entity

import "gorm.io/gorm"

type Chapter struct {
	gorm.Model
	Name        string `json:"name"`
	Description string `json:"description"`
	Order       int    `json:"order" gorm:"index:idx_order_chapter"`

	CourseId uint `json:"courseId" gorm:"index:idx_order_chapter"`

	Course   *Course   `json:"course" gorm:"foreignKey:CourseId; constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Lessions []Lession `json:"lessions" gorm:"foreignKey:ChapterId"`
}

File: e-elearning-main/h-learning-fe-admin/src/model/chapter.ts
import { BaseModel } from "./base";
import { CourseModel } from "./course";
import { LessionModel } from "./lession";

export type ChapterModel = BaseModel & {
    name: string
    description: string
    order: number
    courseId: number

    course?: CourseModel
    lessions: LessionModel[]
}